src/scalarstats.jl

```diff
@@ -37,6 +37,26 @@ function harmmean(a::RealArray)
     return n / s
 end
 
+# Generalized mean
+"""
+    genmean(a, p)
+Return the generalized/power mean with exponent p of a real-valued array.
+"""
+function genmean(a::RealArray, p::Real)
+  if p<0
+    p = convert(AbstractFloat, p)# Negative int p causes domain error.
```

Comment:
More precisely, to ensure type-stability, you need to always use a float when `p` is an integer and `a` holds integers, even when `p` is positive. So you can get rid of that `if`, and always do `pf = convert(AbstractFloat, p)` (and only work with `pf` below).